app/src/main/java/de/danoeh/antennapod/adapter/SelectableAdapter.java

```diff
@@ -156,19 +158,32 @@ private void updateTitle() {
                 selectedIds.size(), getItemCount()));
     }
 
-    public void setOnSelectModeListener(OnSelectModeListener onSelectModeListener) {
-        this.onSelectModeListener = onSelectModeListener;
+    public void setOnStartSelectModeListener(OnStartSelectModeListener onStartSelecModeListener) {
+        this.onStartSelecModeListener = onStartSelecModeListener;
     }
 
+    public void setOnEndSelectModeListener(OnEndSelectModeListener onEndSelectModeListener) {
+        this.onEndSelectModeListener = onEndSelectModeListener;
+    }
+
+
     private void callOnEndSelectMode() {
-        if (onSelectModeListener != null) {
-            onSelectModeListener.onEndSelectMode();
+        if (onEndSelectModeListener != null) {
+            onEndSelectModeListener.onEndSelectMode();
         }
     }
 
-    public interface OnSelectModeListener {
+    private void callOnStartSelectMode() {
```

Comment:
Why did you change this to 2 interfaces? I think that looks less clean (and can be done with standard interfaces like runnable). You always use both of them anyway, so please revert

Reply:
Standard interfaces? Do you mean consumer?

Reply:
Yeah, but please revert back to the single interface like it was before. This also reduces the diff because you don't need to change the other screens

Reply:
Not sure how to use consumer in this case. But I did replace the interfaces with a single interface.